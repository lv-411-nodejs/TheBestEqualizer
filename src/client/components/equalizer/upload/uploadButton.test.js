import React from 'react';
import { shallow } from 'enzyme';
import {
    storeFactory,
  } from '../../../test/testUtils';

import UploadButton from './uploadButton';

jest.mock('pizzicato', () => {
    const mockPizzicato = jest.fn();
    mockPizzicato.context = jest.fn(() => ({}));
    mockPizzicato.context.createAnalyser = jest.fn(() => 'analyser');
    mockPizzicato.Effects = jest.fn(() => {});
    const effects = ['Delay', 'PingPongDelay', 'DubDelay', 'Distortion', 'Quadrafuzz', 'Flanger',
      'Reverb', 'Tremolo', 'StereoPanner', 'Compressor', 'LowPassFilter', 'HighPassFilter', 'RingModulator',
    ];
    effects.forEach((effect, el) => {
      mockPizzicato.Effects[effect] = jest.fn(() => function Effect() {
        return {
          effect: () => (`fakeEffect${el}`),
        };
      });
    });
    return {
      __esModule: true,
      default: mockPizzicato,
    };
  });

const props ={handleInfoFromSound: jest.fn() } 
const handleInfoFromSound = jest.fn(); 

const setup = (initialState = {}) => {
    const store = storeFactory(initialState);
    const wrapper = shallow(<UploadButton handleInfoFromSound={handleInfoFromSound} store={store}/>).debug();
    // wrapper.setProps({props: {handleInfoFromSound: jest.fn()}})
    return wrapper;
  };

describe('test infoAboutUploadFile component', () => {
  let wrapper;
  beforeEach(()=>{
    wrapper = setup();    
  }); 

  it('should render component properly', () => {
    console.log(wrapper)
    expect(wrapper).toMatchSnapshot();
  });
});
