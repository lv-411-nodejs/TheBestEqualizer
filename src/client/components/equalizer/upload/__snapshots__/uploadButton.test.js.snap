// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test infoAboutUploadFile component should render component properly 1`] = `"<UploadButton handleInfoFromSound={[Function: mockConstructor]} store={{...}} dispatch={[Function]} />"`;
